### PR TITLE
Add mechanism for adding extension catalog image to rancher-images.txt

### PR DESCRIPTION
**Pull request by jordojordo**
## Issue: #42092 
`SURE-6065`

***Note***
`ui-plugin-catalog` image can be found on [Docker Hub](https://hub.docker.com/r/rancher/ui-plugin-catalog/tags)

 
## Problem

In order for a user to add any of the UI extensions into an air-gapped instance of Rancher, they will need to manually mirror the image from the [ui-plugin-charts repository](https://github.com/rancher/ui-plugin-charts) into an accessible registry.

## Solution

This adds a mechanism to the `image` package to fetch the latest release from the ui-plugin-charts repo and add the catalog image to the `rancher-images.txt` as well as the `rancher-images-sources.txt` and `rancher-images-origins.txt` files. This mechanism has the ability to add more images to the extension catalog list which may include images from the [partner-extensions repo](https://github.com/rancher/partner-extensions) in the future.

## Testing

Follow the steps in the [wiki](https://github.com/rancher/rancher/wiki/Generating-Image-Lists-Locally-for-Airgap-and-More) for generating the image lists locally.

### Manual Testing

I ran the script mentioned above to test that the image was correctly fetched and applied to the relevant files.

### Automated Testing

* Test types added/modified:
    * Unit
  

Summary: _TODO_

## QA Testing Considerations

 Ensure that the image is correctly formatted and that it does not interfere with other images.
 
### Regressions Considerations

_TODO_

Existing / newly added automated tests that provide evidence there are no regressions:
* _TODO_

### CODE REVIEW

File: pkg/image/extensions.go [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+func ExtensionConfig(endpointIndex int) Extensions {
+	if endpointIndex >= len(ExtensionEndpoints) {
+		endpointIndex = 0
+	}
+
+	return Extensions{
+		Config:         ExportConfig{},
+		GithubEndpoint: ExtensionEndpoints[endpointIndex],
+	}
+}
```
Comment (jiaqiluo) [inline pkg/image/extensions.go:38]: I feel this extra layer of abstraction is unnecessary, and it is confusing to use `ExtensionConfig(0)` to get the hardcoded GithubEndpoint for ui-plugin-charts; what if someone adds a new entry to the beginning of the list ExtensionEndpoints? The code still works but returns a wrong result. 

My suggestion is: 
1. Drop the type `ExtensionEndpoints` and the function `ExtensionConfig` 
2. Make a new `Extensions` instance in reslove.go directly, something like this 
```
	// fetch images from extension catalog images
	extensions := ExtensionConfig{... ... }
	if err := extensions.FetchExtensionImages(imagesSet); err != nil {
		return nil, nil, errors.Wrap(err, "failed to fetch images from extensions")
	}
```

File: pkg/image/extensions.go [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/Masterminds/semver/v3"
+)
+
+type Extensions struct {
+	Config         ExportConfig
```
Comment (jiaqiluo) [inline pkg/image/extensions.go:13]: I do not see any usage of this field. We can consider dropping it unless there is something I do not realize.

**Review by jiaqiluo — CHANGES_REQUESTED** [review]
left a few comments.

File: pkg/image/extensions.go (hunk 1 above)
Comment (jordojordo) [reply pkg/image/extensions.go]: Good points. I removed the type and function, and fixed the `FetchExtensionImages` to iterate correctly through the `ExtensionEndpoints` array.

Let me know what you think.

**Review by jiaqiluo — APPROVED** [review]
LGTM 💯

**Review by rohitsakala — APPROVED** [review]

**Pull request merged**